### DISCUSSION TITLE
Way to locate all broken notebook links?

### DISCUSSION THREAD

**Discussion by aspencoyle** [RobertsLab/resources#1121] [Q&A] [answered]
Over the course of my _Hematodinium_ analysis project, I've learned a lot about how to structure projects, and so have done a lot of restructuring. Upside: My repo is less of a confusing mishmash! Downside: A lot of the links in my lab notebook are no longer functional.

Since I really don't have _that_ many notebook posts, I could probably go through them individually to locate and fix the broken links. However, if there's some non-manual way to do this, I figured it might be good to learn now (and will likely be useful in the future!)

Basically, is there a way to automatically find all the broken links in all my notebook posts?

**Comment by sr320** [comment, marked as answer]
This looks promising -
https://github.com/marketplace/actions/markdown-link-check

**Comment by kubu4** [comment]
I use this, in this location on my computer:

`RobertsLab/sams-notebook/.github/workflows/urlchecker-md-files.yml`

Here's what that is:

```
name: URLChecker
on: [push]

jobs:
  check-urls:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: urlchecker-action
      uses: urlstechie/urlchecker-action@0.2.1
      with:
        # A subfolder or path to navigate to in the present or cloned repository
        subfolder: _posts
        file_types: .md
```

It's a GitHub action and checks URLs each time you push to GitHub.

**Comment by aspencoyle** [comment]
Fantastic, thanks a ton to both of you!!
